### PR TITLE
Send Reminders to SPG Estab Type Military SFA

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
Need to be able to target Military SFA type SPGs with reminders

# What has changed
New estab type included in sample generator
New validator
Test for validator

# How to test?
Load a sample with "Military SFA" as ESTAB_TYPE and check file generates with no errors

# Links
https://trello.com/c/zxsbB9uO/985-reminders-send-reminders-to-spg-estabtype-military-sfa-5
https://collaborate2.ons.gov.uk/confluence/pages/viewpage.action?spaceKey=CRM&title=ESTAB_TYPES+2021
https://collaborate2.ons.gov.uk/confluence/display/SDC/RMC-293+-+Print+reminder+letters+and+questionnaires

### CODE REVIEW

**Comment by LeoKHoward** [comment]
[Reminders - Send reminders to SPG ESTAB_TYPE “MILITARY SFA” (5)](https://trello.com/c/zxsbB9uO/985-reminders-send-reminders-to-spg-estabtype-military-sfa-5)

File: validators.py [hunk 1]
```diff
@@ -107,3 +107,22 @@ def validate(expected_capacity, **kwargs):
...
+                'Hotel',
+                'Hostel',
+                'Residential Caravanner',
+                'Gypsy Roma Traveller',
+                'Residential Boater',
+                'Military SFA']
+
+    def validate(estab_type, **_kwargs):
+        if estab_type not in CE_TYPES and estab_type != 'Household':
+            raise Invalid(f'Estab Type "{estab_type}" is not a valid option')
```
Comment (AdamHawtin) [inline validators.py]: Could add `'Household'` to that list of types then use the existing `in_set` validator instead.

**Review by AdamHawtin — CHANGES_REQUESTED** [review]

File: validators.py [hunk 2]
```diff
@@ -107,3 +107,22 @@ def validate(expected_capacity, **kwargs):
                 f'CE Unit Expected Capacity "{expected_capacity}" cannot be null, blank or zero')
 
     return validate
+
```
Comment (NickGrantX) [inline validators.py]: I don't think you should leave a massive comment block in the code

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Remove comment block

File: validators.py (hunk 2 above)
Comment (LeoKHoward) [reply validators.py]: Good spot

**Review by NickGrantX — APPROVED** [review]

File: validate_sample.py [hunk 3]
```diff
@@ -18,12 +18,15 @@ class SampleValidator:
         'HH_QF3R3AW', 'HH_QFNR1W', 'HH_QFNR2W', 'HH_QFNR3AW', 'HH_3QSFN', 'SPG_QDHSE', 'SPG_QDHSW', 'SPG_LPHUE',
         'SPG_QDHUE', 'SPG_VDNEE', 'CE_QDIEE', 'SPG_VDNEW', 'CE_QDIEW', 'SPG_LPHUW', 'SPG_QDHUW', 'CE_LDIUE', 'CE_LDIUW'}
 
+    ESTAB_TYPES = {'Sheltered Accommodation', 'Hall of Residence', 'Care Home', 'Boarding School', 'Hotel', 'Hostel',
```
Comment (AdamHawtin) [inline validate_sample.py]: Doesn't match the list of estab types in confluence

File: generate_sample_file.py [hunk 4]
```diff
@@ -25,7 +25,8 @@ class SampleGenerator:
                 'Hostel',
                 'Residential Caravanner',
                 'Gypsy Roma Traveller',
-                'Residential Boater']
+                'Residential Boater',
+                'Military SFA']
```
Comment (AdamHawtin) [inline generate_sample_file.py]: Doesn't match the list of estab types in confluence

**Review by AdamHawtin — CHANGES_REQUESTED** [review]
The estab type lists need to match what is documented in confluence attached to the ticket.

**Review by AdamHawtin — APPROVED** [review]

**Pull request merged**